src/concrete/ml/quantization/base_quantized_op.py

```diff
@@ -933,39 +933,62 @@ def make_output_quant_parameters(
         )
 
     def cnp_round(
-        self, x: Union[numpy.ndarray, fhe.tracing.Tracer], calibrate_rounding: bool
+        self,
+        x: Union[numpy.ndarray, fhe.tracing.Tracer],
+        calibrate_rounding: bool,
+        key: Optional[str] = None,
     ) -> numpy.ndarray:
         """Round the input array to the specified number of bits.
 
         Args:
             x (Union[numpy.ndarray, fhe.tracing.Tracer]): The input array to be rounded.
             calibrate_rounding (bool): Whether to calibrate the rounding
                 (compute the lsbs_to_remove)
+            key (Optional[str]): The key to access the appropriate
+                lsbs_to_remove value in the dictionary.
 
         Returns:
             numpy.ndarray: The rounded array.
         """
 
-        # Rounding is applied only if specified by user
-        if self.rounding_threshold_bits is not None:
-            if calibrate_rounding:
-                assert_true(
-                    not isinstance(x, fhe.tracing.Tracer),
-                    "Can't compute lsbs_to_remove at compilation time.",
-                )
-                assert_true(
-                    self.lsbs_to_remove is None,
-                    "Rounding has already been calibrated.",
-                )
+        # Initialize lsbs_to_remove if not already set
+        if not hasattr(self, "lsbs_to_remove") or self.lsbs_to_remove is None:
+            self.lsbs_to_remove = {} if key else 0
```

Comment:
not sure I like the fact that we constantly handle either an int or a dict honestly, couldn't we just make it a dict in any case ( with a default "key" when considering the "single int" case) ?

Reply:
Yes I can probably have a default name that is used when key is not passed.

Reply:
Actually no it makes the qnn tests break...

Reply:
what's the issue ?

Reply:
yes but that could be easily changed no ? like testing we have the specific key instead of an int

not saying it's the best solution though, but I feel like handling the same attribute with two quite different types brings some avoidable complexity tothe base code

Reply:
Alright I made it a dict only.

Reply:
thanks a lot ! I think you still need to update the typing in the class' definition :`    lsbs_to_remove: Optional[Union[int, dict]] = None` (should not be a union anymore)

but apart from that awesome

Reply:
Nice catch. I will update before the merge.